app/helpers/application_helper.rb

```diff
@@ -766,6 +766,23 @@ def payment_settings_url(gateway_type, person, url_params)
     end
   end
 
+  def external_plan_service_login_url(marketplace_id)
+    payload = {user_id: marketplace_id}
+    secret = APP_CONFIG.external_plan_service_jwt_secret
+    external_plan_service_url = APP_CONFIG.external_plan_service_url + "login"
+    token = JWTUtils.encode(payload, secret)
+    URLUtils.append_query_param(external_plan_service_url, "token", token)
+  end
+
+  def display_expiration_notice?
+    if feature_enabled?(:new_plan_page)
+      display = APP_CONFIG.external_plan_service_jwt_secret && @current_user && @current_user.has_admin_rights_in?(@current_community) && PlanUtils.expired?(@current_plan)
+      @external_plan_service_link = external_plan_service_login_url(@current_community.id) if display
```

Comment:
Should this be passed through controller to haml instead of globals (i.e. a bit by bit remove all the globals away)